src/interactions/brush.js

```diff
@@ -1,31 +1,40 @@
 import {create} from "../context.js";
 import {brush as brusher, brushX as brusherX, brushY as brusherY} from "d3";
 import {composeRender} from "../mark.js";
+import {take} from "../options.js";
 
 const states = new WeakMap();
 
 function brushTransform(mode, options) {
   return {
     ...options,
-    // Unlike other composed transforms, interactive transforms must be the
-    // outermost render function because they will re-render dynamically in
-    // response to pointer events.
+    // Interactive transforms must be the outermost render function because they
+    // will re-render dynamically in response to pointer events.
     render: composeRender(function (index, scales, values, dimensions, context, next) {
       const svg = context.ownerSVGElement;
       const {data} = context.getMarkState(this);
 
       // Isolate state per-brush, per-plot; if the brush is reused by multiple
       // marks, they will share the same state.
       let state = states.get(svg);
-      if (!state) states.set(svg, (state = {brushes: [], selection: null}));
-      const {brushes} = state;
+      if (!state) states.set(svg, (state = {brushes: [], bounds: new WeakMap(), selection: null}));
```

Comment:
This doesn’t need to be a WeakMap because the state itself is already weakly held.

```suggestion
      if (!state) states.set(svg, (state = {brushes: [], bounds: new Map(), selection: null}));
```